### PR TITLE
GCS_MAVLink: add mavlink command option to create a 250ms long loop

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

**Comment by peterbarker** [comment]
Tested on a SwifyflyWing thingy

**Comment by peterbarker** [comment]
MAVProxy PR is here: https://github.com/ArduPilot/MAVProxy/pull/853

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 1]
```diff
@@ -2712,6 +2712,15 @@ MAV_RESULT GCS_MAVLINK::handle_preflight_reboot(const mavlink_command_long_t &pa
             param_storage.write_block(0, zeros, sizeof(zeros));
             return MAV_RESULT_FAILED;
         }
+        if (is_equal(packet.param4, 97.0f)) {
+            // create a really long loop
+            send_text(MAV_SEVERITY_WARNING,"Creating long loop");
+            // 250ms:
+            for (uint8_t i=0; i< 250; i++) {
```
Comment (khancyr) [inline libraries/GCS_MAVLink/GCS_Common.cpp:2719]: i < 250

**Review by khancyr — APPROVED** [review]
I assume that is to mess up with the scheduler or something close ?

But it looks good

**Comment by muramura** [comment]
I do not see the value of parameter 4 listed in the MAVLINK specification.
I think it needs to be posted in the ArduPilot WIKI.

https://mavlink.io/en/messages/common.html#MAV_CMD_PREFLIGHT_REBOOT_SHUTDOWN

**Comment by peterbarker** [comment]
@muramura interesting thought.  These are really, *really* in the realm of "ArduPilot internal functions triggerable via mavlink".  

They're ArduPilot developer options - do you really think we should document them as part of the mavlink spec?

**Pull request merged**